direct/io-bulkfs-parquet/src/main/java/cz/o2/proxima/direct/bulk/fs/parquet/ParquetFileFormat.java

```diff
@@ -0,0 +1,189 @@
+import cz.o2.proxima.repository.EntityDescriptor;
+import cz.o2.proxima.storage.StreamElement;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.parquet.hadoop.ParquetOutputFormat;
+import org.apache.parquet.hadoop.metadata.CompressionCodecName;
+import org.apache.parquet.schema.MessageType;
+
+/** Bulk file format which stored data in Parquet format. */
+@Internal
+@Slf4j
+public class ParquetFileFormat implements FileFormat {
+
+  /** Config key specified compression used in parquet writer. */
+  public static final String PARQUET_CONFIG_COMPRESSION_KEY_NAME = ParquetOutputFormat.COMPRESSION;
+
+  /** Config key specified page and row group size in parquet writer. */
+  public static final String PARQUET_CONFIG_PAGE_SIZE_KEY_NAME = ParquetOutputFormat.PAGE_SIZE;
+
+  /**
+   * Config key used for attribute prefix - useful when attribute name collides with proxima fields
+   * defined as PARQUET_COLUMN_NAME_*
+   */
+  public static final String PARQUET_CONFIG_VALUES_PREFIX_KEY_NAME = "parquet.values.name.prefix";
+
+  /** Name for top level message in parquet schema. */
+  public static final String PARQUET_MESSAGE_NAME = "proxima-bulk";
+
+  /** Default page size used for parquet writer */
+  public static final int PARQUET_DEFAULT_PAGE_SIZE = 8 * 1024 * 1024;
+
+  /** Default attribute prefix. See {@link #PARQUET_CONFIG_VALUES_PREFIX_KEY_NAME} for details. */
+  public static final String PARQUET_DEFAULT_VALUES_NAME_PREFIX = "";
+
+  /** Key column name in parquet file */
+  static final String PARQUET_COLUMN_NAME_KEY = "key";
+
+  /** UUID column name in parquet file */
+  static final String PARQUET_COLUMN_NAME_UUID = "uuid";
+
+  /** Attribute column name in parquet file */
+  static final String PARQUET_COLUMN_NAME_ATTRIBUTE = "attribute";
+
+  /** Attribute prefix column name in parquet file */
+  static final String PARQUET_COLUMN_NAME_ATTRIBUTE_PREFIX = "attribute_prefix";
+
+  /** Timestamp column name in parquet file */
+  static final String PARQUET_COLUMN_NAME_TIMESTAMP = "timestamp";
+
+  /** Operation column name in parquet file. See {@link Operation} for details. */
+  static final String PARQUET_COLUMN_NAME_OPERATION = "operation";
+
+  /** Compression for parquet file */
+  CompressionCodecName parquetCompressionCodec;
+
+  @Nullable private transient MessageType parquetSchema;
+  private AttributeFamilyDescriptor familyDescriptor;
+  private String attributeNamesPrefix;
+
+  @Override
+  public void setup(AttributeFamilyDescriptor family) {
+    familyDescriptor = family;
+    parquetCompressionCodec =
+        CompressionCodecName.fromConf(
+            Optional.ofNullable(family.getCfg().get(PARQUET_CONFIG_COMPRESSION_KEY_NAME))
+                .map(Object::toString)
+                .orElse(
```

Comment:
nit: orElseGet

Reply:
That's is not needed because it is wrapped by `CompressionCodecName.fromConf()` which returns default settings (uncompressed)